Modules/Loadable/Markups/Logic/vtkSlicerMarkupsLogic.cxx

```diff
@@ -1561,3 +1561,63 @@ vtkMRMLMarkupsJsonStorageNode* vtkSlicerMarkupsLogic::AddNewJsonStorageNodeForMa
 {
   return vtkMRMLMarkupsJsonStorageNode::SafeDownCast(this->GetMRMLScene()->AddNewNodeByClass(this->GetJsonStorageNodeClassNameForMarkupsType(markupsType)));
 }
+
+//---------------------------------------------------------------------------
+void vtkSlicerMarkupsLogic::SetMarkup(vtkMRMLMarkupsNode *markupsNode,
+                                      vtkSlicerMarkupsWidget* markupsWidget)
+{
+  if (markupsNode == nullptr)
+    {
+    vtkErrorMacro("RegisterMarkup: Invalid node.");
+    return;
+    }
+
+  if (markupsWidget == nullptr)
+    {
+    vtkErrorMacro("RegisterMarkup: Invalid widget.");
+    return;
+    }
+
+  // Take ownership of widget regardless of the outcome of this function (we will
+  // manage the memory).
+  vtkSmartPointer<vtkMRMLMarkupsNode> node =
+    vtkSmartPointer<vtkMRMLMarkupsNode>::Take(markupsNode);
+
+  vtkSmartPointer<vtkSlicerMarkupsWidget> associatedWidget =
+    vtkSmartPointer<vtkSlicerMarkupsWidget>::Take(markupsWidget);
+
+  // Check that the class is not already registered
+  if (this->MarkupsWidgetsMap.count(markupsNode->GetMarkupName()))
+    {
+    vtkWarningMacro("RegisterMarkup: Markups node " << markupsNode->GetMarkupName() <<
+                    " is already registered.");
+    return;
+    }
+
+  this->MarkupsWidgetsMap[markupsNode->GetMarkupName()] = associatedWidget;
+  this->MarkupsNodesMap[markupsNode->GetMarkupName()] = node;
+}
+
+//----------------------------------------------------------------------------
+vtkSlicerMarkupsWidget* vtkSlicerMarkupsLogic::GetWidgetByMarkupsName(const char* markupName) const
```

Comment:
Markups node already have a "Name", so it is a bit confusing what name you mean here.
It should be something like "Type" instead.